javalib/src/main/scala/java/util/concurrent/atomic/AtomicReferenceArray.scala

```diff
@@ -0,0 +1,43 @@
+package java.util.concurrent.atomic
+
+import scala.collection.mutable.ArrayBuffer
+
+class AtomicReferenceArray[E <: AnyRef](
+    length: Int) extends Serializable {
+
+  def this(array: Array[E]) = {
+    this(array.size)
+    inner = ArrayBuffer[E](array: _*)
+  }
+
+  private var inner: ArrayBuffer[E] = new ArrayBuffer[E](length)
+
+  final def length(): Int = inner.size
```

Comment:
ok, just curious on the difference ...

2015-06-03 22:51 GMT+02:00 Sébastien Doeraene <notifications@github.com>:

> In
> javalib/src/main/scala/java/util/concurrent/atomic/AtomicReferenceArray.scala
> <https://github.com/scala-js/scala-js/pull/1689#discussion_r31668345>:
>
> > @@ -0,0 +1,43 @@
> > +package java.util.concurrent.atomic
> > +
> > +import scala.collection.mutable.ArrayBuffer
> > +
> > +class AtomicReferenceArray[E <: AnyRef](
> > +    length: Int) extends Serializable {
> > +
> > +  def this(array: Array[E]) = {
> > +    this(array.size)
> > +    inner = ArrayBuffer[E](array: _*)
> > +  }
> > +
> > +  private var inner: ArrayBuffer[E] = new ArrayBuffer[E](length)
> > +
> > +  final def length(): Int = inner.size
>
> And then it's preferable to use inner.length here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scala-js/scala-js/pull/1689/files#r31668345>.
>